lib/protobuf/field/bytes_field.rb

```diff
@@ -58,21 +58,15 @@ def define_setter
         message_class.class_eval do
           define_method(method_name) do |val|
             @encode = nil
-            begin
-              case val
-              when String, Symbol
-                @values[field.name] = "#{val}"
-              when NilClass
-                @values.delete(field.name)
-              when ::Protobuf::Message
-                @values[field.name] = val.dup
-              else
-                fail TypeError, "Unacceptable value #{val} for field #{field.name} of type #{field.type_class}"
-              end
-            rescue NoMethodError => ex
```

Comment:
Interesting that we had no test coverage for this behavior. In either case we are raising an exception and we can let the caller decider if they want to log and convert the error type themselves :+1: